includes/api/request.php

```diff
@@ -2,545 +2,255 @@
-
-	public function __construct($domain,$token, $secret, $debug = 0){
-        $this->domain = $domain;	
-        $this->token = $token;
-        $this->secret = $secret;
-        $this->debug = 	$debug;
-        $this->endpoint = 'https://'.strtolower($domain).'.trackhs.com';
-	}
-    
-    public function getEndPoint(){
+    protected $endpoint;
+    protected $client;
+    protected $url;
+
+    private $base_uri;
+    private $public_key;
+    private $private_key;
+
+    public function __construct()
+    {
+        $this->base_uri = env('TRACK_BASE_URI');
+        $this->public_key = env('TRACK_PUBLIC_KEY');
+        $this->private_key = env('TRACK_PRIVATE_KEY');
+    }
+
+    public function request($params = [])
+    {
+        $curl = curl_init();
+        $opts = [
+            CURLOPT_URL => $this->base_uri.$params['endpoint'],
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_ENCODING => "",
+            CURLOPT_MAXREDIRS => 10,
+            CURLOPT_TIMEOUT => 30,
+            CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+            CURLOPT_CUSTOMREQUEST => "GET",
+            CURLOPT_HTTPHEADER => [
+                "Accept: application/json",
+                "Authorization: Basic ".base64_encode($this->public_key.":".$this->private_key)
+            ],
+        ];
+        curl_setopt_array($curl, $opts);
+        $response = curl_exec($curl);
+        $error = curl_error($curl);
+        curl_close($curl);
+
+        return ($error) ? "cURL Error #:$error" : json_decode($response);
+    }
+
+    public function getEndPoint()
+    {
         return $this->endpoint;
     }
-    
-    public function getUnitCount(){
+
+    public function getUnitCount($endpoint = false)
+    {
         global $wpdb;
-        $units = wp_remote_post($this->endpoint.'/api/wordpress/unit-count/',
-		array(
-			'timeout'     => 500,
-            'user-agent' => apply_filters( 'http_headers_useragent', 'WordPress/' . get_bloginfo( 'version' ) . '; TrackConnect/'. WP_LISTINGS_VERSION .'; ' . get_bloginfo( 'url' ) ),
-            'body' => array(
-    			'token'     => $this->token
-			    )
-			)
-        );
-           
-        // Disable inactive units      
-        $ids = json_decode($units['body'])->units;
-        
-        $posts = $wpdb->get_results("SELECT posts.ID AS id FROM `".$wpdb->prefix."posts` AS posts
-            WHERE post_type='listing' AND post_status ='publish'
-            AND ID IN(SELECT post_id FROM `".$wpdb->prefix."postmeta` WHERE meta_key='_listing_unit_id' AND meta_value NOT IN (".implode(',',$ids).") ) ");
-        
-		if(!empty($posts)){
-    		foreach($posts as $post){
-    			$wpdb->update( $wpdb->posts, 
-                    array( 'post_status' => 'draft' ), 
-                    array( 'ID' => $post->id ), 
-                    array( '%s' ), array( '%d' ) );
-            }
+
+        $nextEndpoint = $endpoint ? str_replace($this->base_uri, '', $endpoint) : '/api/pms/units?size=100';
```

Comment:
I see you're setting the limit here to 100 items, but I'm not seeing it get paginated in any place. Can you show me where you're paginating it ?

Reply:
https://github.com/Villatel/track-connect/blob/feature/track_plugin_change_api_endpoints/includes/api/request.php#L75-L77

?size=100 it's only first request, on link above we get next items if they exist. I have limited it to 100 because _all takes too long to respond